lib/board.rb

```diff
@@ -0,0 +1,77 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" "," "," "," "," "," "," "," "," "]
+  end
+
+  def reset!
+    @cells = [" "," "," "," "," "," "," "," "," "]
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(num)
+    num = num.to_i
+    cells[num-1]
+  end
+
+  def full?
+    value = true
+    cells.each do |cell|
+      if cell == " "
+        value = false
+      end
+    end
+    return value
+  end
+
+  def turn_count
+    value = 0
+    cells.each do |cell|
+      if cell != " "
+        value +=1
+      end
+    end
+    return value
+  end
+
+  def taken?(num)
+    value = false
+    if position(num) == "X" || position(num) == "O"
+      value = true
+    end
+    return value
+  end
+
+  def valid_move?(num)
```

Comment:
Use 'and' operator (`&&`) to combine multiple logical expressions and utilize their implicit return value.
```
input.to_i.between?(1,9) && !taken(input)
```